### PR TITLE
[RAPTOR-2697] add hook to read dataset file and move reading into Py/R/Java context

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Wild mode requires that user could provide a hook to read data file; this requires moving file reading into the language(Py/R/Java) specific context.

Hook name is: `read_file`, your are welcome to suggest better namimg.

There is inconsistency regarding reading csv into dataframe.
Pandas dataframe treats missing values as NaN.
R dataframe treats missing values as NA. Replacing NA with NaN for compatibility sake, and  going to consult with CFDS.

### CODE REVIEW

**Comment by yakov-g** [comment]
jarvis test functional_drop_in

**Review by meltzg — CHANGES_REQUESTED** [review]
update env ids?

File: README.md [hunk 1]
```diff
@@ -94,6 +94,11 @@ include any necessary hooks in a file called `custom.py` for Python models or `c
   - Executed once in the beginning of the run
   - `kwargs` - additional keyword arguments to the method;
     - code_dir - code folder passed in the `--code_dir` parameter
+- `read_file(input_filename: str) -> Any`
+  - `input_filename` is a data file, passed in the `--input` parameter
+  - If used, this hook must return a non-None value
```
[the remark this answers is not in the document]
Comment (yakov-g) [reply README.md:99]: done

File: README.md [hunk 2]
```diff
@@ -94,6 +94,11 @@ include any necessary hooks in a file called `custom.py` for Python models or `c
   - Executed once in the beginning of the run
   - `kwargs` - additional keyword arguments to the method;
     - code_dir - code folder passed in the `--code_dir` parameter
+- `read_file(input_filename: str) -> Any`
```
[the remark this answers is not in the document]
Comment (yakov-g) [reply README.md]: yes, sounds better.
fixed

**Comment by 4dahalibut** [comment]
can you name it read_data or read_csv

**Review by meltzg — APPROVED** [review]

**Comment by yakov-g** [comment]
> can you name it read_data or read_csv
It can not be `read_csv` as in wild model input can be anything.

@meltzg @4dahalibut So either `read_input` or `read_data` ?

`read_input` is very clear but not very nice...
`read_data` is nicer but less clear

**Comment by 4dahalibut** [comment]
read_input_data then maybe.

**Comment by meltzg** [comment]
i like `read_input` because it matches the `--input` flag

**Comment by yakov-g** [comment]
I'd also choose `read_input_data`.
Do we want to ask on linguists forum?

**Comment by yakov-g** [comment]
Let's do `read_input_data`
there is some `read_input` method in R https://www.rdocumentation.org/packages/pastis/versions/0.1-2/topics/read_input

And we also, might need to rename `init` as I saw some warning that it clashes with R init

**Comment by yakov-g** [comment]
We may also add `drcm` prefix to the hooks to avoid clashing.

**Review by 4dahalibut — APPROVED** [review]

**Comment by datarobot-jarvis** [comment]
:dollar: Total cost of executing tests in Jenkins: $**9.18** USD.

[What is this all about?](http://bit.ly/2BahK4V)

**Pull request merged**